addons/web/static/src/js/views/list/list_controller.js

```diff
@@ -129,6 +129,23 @@ var ListController = BasicController.extend({
             return self.model.get(db_id, {raw: true});
         });
     },
+    /**
+     * Returns the list of currently selected records (with the check boxes on
+     * the left) or the whole domain records if it is selected
+     *
+     * @returns {Promise<{id, display_name}[]>}
+     */
+    getSelectedRecordsWithDomain: async function () {
+        if (this.isDomainSelected) {
+            const state = this.model.get(this.handle, {raw: true});
+            return await this._domainToRecords(state.getDomain(), session.active_ids_limit);
+        } else {
+            return Promise.resolve(this.selectedRecords.map(localId => {
+                const data = this.model.localData[localId].data;
+                return { id: data.id, display_name: data.display_name };
```

Comment:
I'm not sure "display_name" is always present in the data. But maybe we can live with the limitation for now. It means we have to put the field in the list view's arch.